### ISSUE TITLE
Idea: Add configure option to require simdjson PECL for faster json decoding

### ISSUE THREAD

**Issue by TysonAndre** [phpredis/phpredis#2193]
### Expected behaviour

See https://github.com/crazyxman/simdjson_php/pull/86 for an example of how to call it - this would work when `defined(SIMDJSON_PHP_VERSION_ID) && SIMDJSON_PHP_VERSION_ID >= 30000`

- Note that the C library internally used by the pecl uses 32-bit offset types, and will reject strings exceeding 4GB, but "Bulk Strings are used in order to represent a single binary-safe string up to 512 MB in length."

https://github.com/crazyxman/simdjson_php/pull/86/files#diff-e40ddac25bf760132109ead4b3fd8bbb2ab571a5ed01d7eb841a2c81bf8d28d3

For valid json, simdjson should have equivalent behavior to php's json decoder, and simdjson will return a non-zero error code when the depth is too large or the json is invalid

For decoding json, this is faster than php's bison-based json parser in various use cases: https://github.com/crazyxman/simdjson_php/tree/master/benchmark#run-benchmark

Is this something you'd be interested in a PR for? (e.g. `configure --enable-redis-simdjson`)

### I'm seeing this behaviour on
- OS: N/A
- Redis: N/A
- PHP: N/A
- phpredis: N/A

### I've checked
- [x] There is no similar issue from other users
- [x] Issue isn't fixed in `develop` branch

**Comment by michael-grunder** [comment]
I agree that it would be very cool.  Honestly, I've wanted an excuse to use simdjson since I read Lemire's blog post about it a few years back. 😂
